backend/server.js

```diff
@@ -18,12 +44,54 @@ const app = express()
 app.use(cors())
 app.use(bodyParser.json())
 
+const authenticateUser = async (req, res, next) => {
+  const user = await User.findOne({ accessToken: req.header('Authorization') })
+  if (user) {
+    req.user = user //what does this mean? 
+    next() //when to use next? (calling the next() function which allows the proteced endpoint to continue execution)
+  } else {
+    res.status(403).json({ message: "You need to login to access this page" })
+  }
+}
+
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Hello backend world!!!')
+})
+
+// Create user
+// Database only saves correct users, but doesn't display error when incorrect users are posted / Fixed
+app.post('/users', async (req, res) => {
+  try {
+    const { name, email, password } = req.body
+    // What happens here???
```

Comment:
don't know what you mean with this question, but you're destructuring name, email and password from the req.body. Then you're creating a new user based on you User model and encrypting the password, so it's not saved as plain text. then your waiting on it to be saved in the db and then your sending that saved user back as the res.json()